tests/ArmTemplates.Tests/Extractor/Scenarios/PolicyExtractorTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="PolicyExtractorTests.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Executors;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Policy;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Extractor.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Moqs.ApiClients;
+using Xunit;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Extractor.Scenarios
+{
+    [Trait("Category", "Policy Extraction")]
+    public class PolicyExtractorTests : ExtractorTestsBase
+    {
+        static string OutputPoliciesDirectory;
+
+        public PolicyExtractorTests()
+        {
+            OutputPoliciesDirectory = Path.Combine(TESTS_OUTPUT_DIRECTORY, "policy-tests");
+
+            // remember to clean up the output directory before each test
```

Comment:
Isn't this ran only once and not before each individual test?

Reply:
Extractor generates files. I decided to get such a folder structure in tests:
```
tests-output\
    policy-tests\
        <name_of_test>\
            files
            files
            files
        <name_of_test>\
            files
            files
            files
```

So it is up to a developer to clear tests output directory or not - if you are running some configuration tests, then you maybe don't want to clear all the tests-output directory (it is slowing down your tests launches).

So what is why i decided to call delete here - before each test in this class (class is for policy tests)